jax/_src/numpy/lax_numpy.py

```diff
@@ -1005,10 +1005,24 @@ def sinc(x):
   _check_arraylike("sinc", x)
   x, = _promote_dtypes_inexact(x)
   eq_zero = lax.eq(x, lax._const(x, 0))
-  safe_x = where(eq_zero, lax._const(x, 0), x)
-  pi_x = lax.mul(lax._const(x, pi), safe_x)
-  return where(eq_zero,
-               lax._const(x, 1), lax.div(lax.sin(pi_x), pi_x))
+  pi_x = lax.mul(lax._const(x, pi), x)
+  safe_pi_x = where(eq_zero, lax._const(x, 0), pi_x)
+  return where(eq_zero, _sinc_maclaurin(0, pi_x),
+               lax.div(lax.sin(safe_pi_x), safe_pi_x))
+
+@partial(custom_jvp, nondiff_argnums=(0,))
+def _sinc_maclaurin(k, x):
+  # compute the kth derivative of x -> sin(x)/x evaluated at zero (since we
+  # compute the monomial term in the jvp rule)
+  if k % 2:
+    return lax._const(x, 0)
+  else:
+    return lax._const(x, (-1) ** (k // 2) / (k + 1))
```

Comment:
Actually, isn't this missing a factorial in the denominator?

Reply:
Great thing to check! It's always confusing to me where the factorials live. The only way I'm ever confident is by checking numerically. (And perhaps the function name could be chosen better; see below.)

I _think_ the current expression, without a factorial, is right:

```python
In [1]: from jax import grad

In [2]: import jax.numpy as jnp

In [3]: f = jnp.sinc

In [4]: for i in range(10):
   ...:     f = grad(f)
   ...:

In [5]: f(0.)  # slow! jax.experimental.jet would be much faster
WARNING:absl:No GPU/TPU found, falling back to CPU. (Set TF_CPP_MIN_LOG_LEVEL=0 and rerun for more info.)
Out[5]: DeviceArray(-8513.462, dtype=float32)
```

That agrees with [Wolfram Alpha's symbolic answer](https://www.wolframalpha.com/input/?i=10th+derivative+of+sin%28pi+*+x%29+%2F+%28pi+*+x%29+at+0) of `-jnp.pi ** 10 / 11`.

Another way to check is to make sure the derivatives are all continuous at zero:

```python
import matplotlib.pyplot as plt
from jax import grad, vmap
import jax.numpy as jnp


xs = jnp.linspace(-5, 5, 101)
plt.plot(xs, vmap(grad(jnp.sinc))(xs))
plt.plot(xs, vmap(grad(grad(jnp.sinc)))(xs))
plt.plot(xs, vmap(grad(grad(grad(jnp.sinc))))(xs))
plt.plot(xs, vmap(grad(grad(grad(grad(jnp.sinc)))))(xs))
plt.savefig('sinc_4th.png')
```

![sinc_4th](https://user-images.githubusercontent.com/1458824/100904138-91ff9500-347b-11eb-958e-a3d7c9953c86.png)

(Interestingly, we can't use `jnp.linspace(-5, 5, 1001)` because for small nonzero values our derivatives get numerically unstable... maybe I'll use the Maclaurin series approximation _near_ zero, not just _at_ zero.)

(Also, this test caught a bug where I wasn't broadcasting arrays correctly!)

About the function name, this function is really computing the kth _derivative_ of sinc at zero, not the kth Taylor/Maclaurin coefficient (which is the kth derivative divided by k factorial). The function might instead be named `_deriv_of_sinc_at_zero`, at least in terms of describing the primal value that gets computed. But I renamed it to `_sinc_maclaurin` because the way it behaves under autodiff makes it act to its caller more like an accurate-as-needed Taylor expansion of sinc at zero.